app/src/main/java/com/simplecity/amp_library/playback/MusicService.java

```diff
@@ -1581,28 +1674,100 @@ public void moveQueueItem(int from, int to) {
         }
     }
 
-    public void makeShuffleList() {
+    public void makeTrackShuffleList() {
         synchronized (this) {
 
             if (playlist == null || playlist.isEmpty()) {
                 return;
             }
 
-            shuffleList = new ArrayList<>(playlist);
+            trackShuffleList = new ArrayList<>(playlist);
             Song currentSong = null;
-            if (playPos >= 0 && playPos < shuffleList.size()) {
-                currentSong = shuffleList.remove(playPos);
+            if (playPos >= 0 && playPos < trackShuffleList.size()) {
+                currentSong = trackShuffleList.remove(playPos);
             }
 
-            Collections.shuffle(shuffleList);
+            Collections.shuffle(trackShuffleList);
 
             if (currentSong != null) {
-                shuffleList.add(0, currentSong);
+                trackShuffleList.add(0, currentSong);
             }
             playPos = 0;
         }
     }
 
+    public void makeAlbumShuffleList() {
+        synchronized (this) {
+            if (albumShuffleList.size() > 0) {
```

Comment:
Unless we need to init the `albumShuffleList` here even in early-return cases, I would put this after the early-return (`playlist == null`) check. Just a code style thing - it reads better to me if early return code happens first.